Mathlib/Algebra/Group/Translate.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.BigOperators.Pi
+import Mathlib.Algebra.Group.Pi.Basic
+import Mathlib.Data.Set.Pointwise.SMul
+import Mathlib.GroupTheory.GroupAction.DomAct.Basic
+
+/-!
+# Translation operator
+
+This file defines the translation of a function from a group by an element of that group.
+
+## Notation
+
+`τ a f` is notation for `translate a f`.
+
+## See also
+
+Generally, translation is the same as acting on the domain by subtraction. This setting is
+abstracted by `DomAddAct` in such a way that `τ a f = DomAddAct.mk (-a) +ᵥ f` (see
+`translate_eq_domAddActMk_vadd`). Using `DomAddAct` is irritating in applications because of this
+negation appearing inside `DomAddAct.mk`. Although mathematically equivalent, the pen and paper
+convention is that translating is an action by subtraction, not by addition.
+-/
+
+open Function Set
+open scoped Pointwise
+
+variable {ι α β M G H : Type*} [AddCommGroup G]
+
+/-- Translation of a function in a group by an element of that group.
+`τ a f` is defined as `x ↦ f (x - a)`.  -/
+def translate (a : G) (f : G → α) : G → α := fun x ↦ f (x - a)
```

Comment:
Should we multiplicitivize this already? I wouldn't be surprised if we want to use this for `Units Complex` or `GL n` or something like that.

Reply:
What name would you like me to use?

Reply:
`leftMulTranslate`? Would that work?